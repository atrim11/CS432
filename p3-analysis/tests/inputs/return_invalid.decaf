// Test: Void function returning a value
def int voidFunction() {
    return true; // Error: void function cannot return a value
}

def int main() {
    return voidFunction();
}