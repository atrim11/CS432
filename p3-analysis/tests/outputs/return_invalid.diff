--- outputs/return_invalid.txt	2024-10-23 10:10:39.594556275 -0400
+++ expected/return_invalid.txt	2024-10-22 15:42:36.467384468 -0400
@@ -1,20 +1 @@
-Program [line 1]
-SYM TABLE:
- print_int : (int) -> void
- print_bool : (bool) -> void
- print_str : (str) -> void
- voidFunction : () -> void
- main : () -> void
-
-  FuncDecl name="voidFunction" return_type=void parameters={} [line 2]
-  SYM TABLE:
-
-    Block [line 2]
-    SYM TABLE:
-
-  FuncDecl name="main" return_type=void parameters={} [line 6]
-  SYM TABLE:
-
-    Block [line 6]
-    SYM TABLE:
-
+Type mismatch: function return type is bool but declared as int on line 1
