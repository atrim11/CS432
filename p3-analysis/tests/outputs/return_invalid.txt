Main function must return an int on line 6
