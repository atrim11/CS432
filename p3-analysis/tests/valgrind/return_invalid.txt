==290678== Memcheck, a memory error detector
==290678== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==290678== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==290678== Command: ../decaf inputs/return_invalid.decaf
==290678== 
Type mismatch in return statement on line 3: expected int, got bool
==290678== 
==290678== HEAP SUMMARY:
==290678==     in use at exit: 0 bytes in 0 blocks
==290678==   total heap usage: 1,818 allocs, 1,818 frees, 222,532 bytes allocated
==290678== 
==290678== All heap blocks were freed -- no leaks are possible
==290678== 
==290678== For lists of detected and suppressed errors, rerun with: -s
==290678== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
