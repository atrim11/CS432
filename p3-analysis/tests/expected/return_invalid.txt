Type mismatch in return statement on line 3: expected int, got bool
